test/dd/test_package.cpp

```diff
@@ -2165,6 +2176,34 @@ TEST(DDPackageTest, InnerProductTopNodeConjugation) {
   EXPECT_NEAR(dd->expectationValue(op, evolvedState), -0.416, 0.001);
 }
 
+TEST(DDPackageTest, GetMatrixMultiQubitIdentity) {
+  const auto nrQubits = 2U;
+
+  auto dd = std::make_unique<dd::Package<>>(nrQubits);
+
+  auto identity = dd->makeIdent();
+  auto matrix = dd->getMatrix(identity, nrQubits);
+
+  auto goalRow0 = dd::CVec{{1., 0.}, {0., 0.}, {0., 0.}, {0., 0.}};
+  auto goalRow1 = dd::CVec{{0., 0.}, {1., 0.}, {0., 0.}, {0., 0.}};
+  auto goalRow2 = dd::CVec{{0., 0.}, {0., 0.}, {1., 0.}, {0., 0.}};
+  auto goalRow3 = dd::CVec{{0., 0.}, {0., 0.}, {0., 0.}, {1., 0.}};
+  auto goalMatrix = dd::CMat{goalRow0, goalRow1, goalRow2, goalRow3};
+  ASSERT_EQ(dd->getMatrix(identity, nrQubits), goalMatrix);
+}
+
+TEST(DDPackageTest, GetMatrixCNOT) {
+  const auto nrQubits = 2U;
+
+  auto dd = std::make_unique<dd::Package<>>(nrQubits);
+
+  auto originalDD = dd->makeGateDD(dd::X_MAT, 2, 1_pc, 0);
+  auto matrix = dd->getMatrix(originalDD, nrQubits);
+  auto recreatedDD = dd->makeDDFromMatrix(matrix);
+
+  ASSERT_EQ(originalDD, recreatedDD);
+}
```

Comment:
These are just rather rudimentary tests that have been used for developing. Before the PR is being merged, they should be extended to more elaborate tests that explicitly make sure the new functionality works as intended.